lib/rspec/support/object_inspector.rb

```diff
@@ -55,6 +74,25 @@ def self.register(klass, &block)
       def self.registered_klasses
         @registered_klasses ||= {}
       end
+      class ArrayInspector
+        def self.inspect(array)
+          array.map do |element|
+            if Array === element
+              ArrayInspector.inspect(element)
+            elsif Hash === element
+              HashInspector.inspect(element)
+            else
+              ObjectInspector.formatter(element)
+            end
+          end
```

Comment:
Maybe I'm missing something, but can't this method def be simplified to:

``` ruby
def self.inspect(array)
  array.map do |element|
    ObjectInspector.formatter(element)
  end
end
```

After all, `ObjectInspector` takes care of the array and hash cases already.